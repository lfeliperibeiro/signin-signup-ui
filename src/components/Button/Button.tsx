import React from 'react';

interface ButtonProps
  extends React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > {}

export const Button = ({ children, ...rest }: ButtonProps) => {
  return (
    <button
      data-testid={'button'}
      className={
        'text-xl bg-amber-600 text-white w-full px-0.5 py-2 rounded hover:bg-amber-900'
      }
      {...rest}
    >
      {children}
    </button>
  );
};
