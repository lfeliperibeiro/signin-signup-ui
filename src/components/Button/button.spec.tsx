import { render, screen } from '@testing-library/react';
import { Button } from './Button';

describe('Button', () => {
  it('should render correct', () => {
    render(<Button>test</Button>);
    const button = screen.getByTestId('button');
    expect(button).toBeTruthy();
    expect(button).toMatchSnapshot();
  });
});
