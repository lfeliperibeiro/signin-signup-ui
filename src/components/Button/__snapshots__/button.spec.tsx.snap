// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correct 1`] = `
<button
  class="text-xl bg-amber-600 text-white w-full px-0.5 py-2 rounded hover:bg-amber-900"
  data-testid="button"
>
  test
</button>
`;
